### ISSUE TITLE
Change batch_norm naming?

### ISSUE THREAD

**Issue by DomInvivo** [datamol-io/graphium#71]
For GNNs, multiple types of norming have been developed and batch norm is not necessarily the best. We also have graph_norm, layer_norm, instance_norm.

Should we change the naming convention of NN layers? Instead of `batch_norm=BOOL`, we would have `norm=STR`, where the str will be `batch_norm`, `instance_norm`, `graph_norm`, and have a norm parser?

Doing this change after the release might require to have depreciation warnings and stuff like that, so even if none of the other norms are implemented, I think it's a good idea to make the move now. What do you think @hadim ?
